static/js/compose_state.js

```diff
@@ -41,8 +41,36 @@ export const topic = get_or_set("stream_message_recipient_topic");
 // of the indented syntax for multi-line code blocks.
 export const message_content = get_or_set("compose-textarea", true);
 
-export function focus_in_empty_compose() {
-    return composing() && message_content() === "" && $("#compose-textarea").is(":focus");
+export function focus_in_empty_compose(event_name) {
+    // User trying to press arrow keys in an empty compose is mostly
+    // likely trying to navigate messages.
+    if (!composing() || message_content() !== "") {
+        return false;
+    }
+
+    const focused_element_id = document.activeElement.id;
+    if (focused_element_id === "compose-textarea") {
+        return true;
+    }
+
+    if (event_name === "down_arrow") {
+        // If the focus is on recipient boxes and user pressed down arrow,
+        // we should not close the compose box as user may be trying to navigate to
+        // the textarea.
+        return false;
+    }
```

Comment:
I don't understand why we'd want to make that special case here, and not in the compose box in general.  The intent of this feature is to make it not important that you forgot you had an empty compose box open when you're trying to use navigation keys, and I think if we want an exception of this form I'd expect it to apply in all cases.  

So I've removed block.  To be honest, I'm not sure that this will come up in practice at all, but if it does, I think the simpler implementation without the `if` statement is more intuitive, and in any case, it's simpler.